src/main/java/ekud/commands/Command.java

```diff
@@ -0,0 +1,70 @@
+package ekud.commands;
+
+import java.util.HashMap;
+
+import ekud.components.Storage;
+import ekud.components.TaskList;
+import ekud.exceptions.EkudException;
+import ekud.ui.Ui;
+
+/**
+ * Represents commands made by the user.
+ *
+ * @author uniqly
+ */
+public abstract class Command {
+    /**
+     * Represents the types of commands the user can make.
+     */
+    public enum Type {
+        LIST, ADD, DELETE, MARK, UNMARK, EXIT, FIND;
+
+        /** A set of aliases for the different {@link Type Types} of {@link Command} */
+        private static final HashMap<String, Type> aliases;
+        // initialize command aliases
+        static {
+            aliases = new HashMap<>();
+            aliases.put("list", LIST);
+            aliases.put("todo", ADD);
+            aliases.put("deadline", ADD);
+            aliases.put("event", ADD);
+            aliases.put("delete", DELETE);
+            aliases.put("mark", MARK);
+            aliases.put("unmark", UNMARK);
+            aliases.put("bye", EXIT);
+            aliases.put("exit", EXIT);
+            aliases.put("find", FIND);
+        }
+
+        /**
+         * Converts a given {@link String} into a {@link Type}.
```

Comment:
Could the comment for getType method could be clearer? Maybe instead of "Converts a given {@link String} into a {@link Type}.", it might be more descriptive to say "Returns the Command Type associated with the given input string."?